pkg/storegateway/bucket.go

```diff
@@ -909,24 +841,18 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 			stats.mergedSeriesCount += seriesCount
 			stats.mergedChunksCount += chunksCount
 
-			if s.maxSeriesPerBatch <= 0 {
-				// When streaming is disabled, the time spent iterating over the series set is the
-				// actual time spent merging series and sending response to the client.
-				stats.synchronousSeriesMergeDuration += time.Since(iterationBegin)
-			} else {
-				// When streaming is enabled, the time spent iterating over the series set is the
-				// actual time spent fetching series and chunks, encoding and sending them to the client.
-				// We split the timings to have a better view over how time is spent.
-				stats.streamingSeriesFetchSeriesAndChunksDuration += stats.streamingSeriesWaitBatchLoadedDuration
-				stats.streamingSeriesEncodeResponseDuration += encodeDuration
-				stats.streamingSeriesSendResponseDuration += sendDuration
-				stats.streamingSeriesOtherDuration += time.Duration(util_math.Max64(0, int64(time.Since(iterationBegin)-
-					stats.streamingSeriesFetchSeriesAndChunksDuration-encodeDuration-sendDuration)))
-			}
+			// The time spent iterating over the series set is the
+			// actual time spent fetching series and chunks, encoding and sending them to the client.
+			// We split the timings to have a better view over how time is spent.
+			stats.streamingSeriesFetchSeriesAndChunksDuration += stats.streamingSeriesWaitBatchLoadedDuration
+			stats.streamingSeriesEncodeResponseDuration += encodeDuration
+			stats.streamingSeriesSendResponseDuration += sendDuration
+			stats.streamingSeriesOtherDuration += time.Duration(util_math.Max64(0, int64(time.Since(iterationBegin)-
+				stats.streamingSeriesFetchSeriesAndChunksDuration-encodeDuration-sendDuration)))
 		})
 
 		// NOTE: We "carefully" assume series and chunks are sorted within each SeriesSet. This should be guaranteed by
-		// blockSeries method. In worst case deduplication logic won't deduplicate correctly, which will be accounted later.
+		// blockSeriesSkippingChunks method. In worst case deduplication logic won't deduplicate correctly, which will be accounted later.
```

Comment:
If I understand correct blockSeriesSkippingChunks is only used in LabelNames, not Series, so I don't understand this comment.

Reply:
This comment was applying to the old implementation and I forgot to remove it. The new implementation doesn't deduplicate chunks at all (it's delegated to the querier).

I removed this comment in https://github.com/grafana/mimir/pull/4332/commits/ace71e2a8ff4bf0de9abb8080ef84968d3e858ae